### PR TITLE
bpo-43475: Fix the Python implementation of hash of Decimal NaN

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue43475

### CODE REVIEW

File: Lib/test/test_decimal.py [hunk 1]
```diff
@@ -1814,13 +1814,7 @@ def hashit(d):
 
         # check that hash(d) == hash(int(d)) for integral values
         for value in test_values:
-            self.assertEqual(hashit(value), hashit(int(value)))
-
-        #the same hash that to an int
-        self.assertEqual(hashit(Decimal(23)), hashit(23))
```
Comment (serhiy-storchaka) [inline Lib/test/test_decimal.py:1820]: Covered by tests above.

File: Lib/test/test_decimal.py [hunk 2]
```diff
@@ -1814,13 +1814,7 @@ def hashit(d):
...
-        self.assertRaises(TypeError, hash, Decimal('sNaN'))
-        self.assertTrue(hashit(Decimal('Inf')))
-        self.assertTrue(hashit(Decimal('-Inf')))
```
Comment (serhiy-storchaka) [inline Lib/test/test_decimal.py:1823]: Covered by tests below.

File: Lib/test/test_decimal.py [hunk 3]
```diff
@@ -1814,13 +1814,7 @@ def hashit(d):
 
         # check that hash(d) == hash(int(d)) for integral values
         for value in test_values:
-            self.assertEqual(hashit(value), hashit(int(value)))
-
-        #the same hash that to an int
-        self.assertEqual(hashit(Decimal(23)), hashit(23))
-        self.assertRaises(TypeError, hash, Decimal('sNaN'))
```
Comment (serhiy-storchaka) [inline Lib/test/test_decimal.py:1821]: Moved to `test_hash_method_nan`.

**Review by mdickinson — APPROVED** [review]
LGTM

**Pull request merged**